src/pipeline.js

```diff
@@ -226,15 +253,25 @@ class Pipeline {
      * Reduction function used to process the pipeline functions and merge the context parameters.
      * @param {Object} currContext Accumulated context
      * @param {pipelineFunction} currFunction Function that is currently "reduced"
+     * @param {number} index index of the function in the given array
      * @returns {Promise} Promise resolving to the new value of the accumulator
      */
-    const merge = (currContext, currFunction, index) => {
-      // copy the pipeline payload into a new object to avoid modifications
-      const mergedargs = _.merge({}, currContext);
+    const merge = async (currContext, currFunction, index) => {
+      const skip = (!currContext.error) === (!!currFunction.errorHandler);
```

Comment:
As I wrote in #85 I don't think this skipping magic is a good idea. Instead I would check if `context.errors!=null && this._errors.length> 1` (see above).

Reply:
the idea is, that the `error()` can be set like a `when()`:

```
before(fetch)
.error(recover_failed_fetch)
.once(...)
...
```